pkg/apis/gcp/v1alpha1/types_infrastructure.go

```diff
@@ -111,6 +114,26 @@ type CloudNAT struct {
 	// The default value is 2048 ports.
 	// +optional
 	MinPortsPerVM *int32 `json:"minPortsPerVM,omitempty"`
+	// NatIPNames is a list of all user provided external premium ips which can be used by the nat gateway
+	// +optional
+	NatIPNames *[]NatIPName `json:"natIPNames,omitempty"`
+	// NatIPs is a list of all user provided external premium ips which can be used by the nat gateway
+	// +optional
+	NatIPs *[]NatIP `json:"natIPs,omitempty"`
```

Comment:
Can we harmonize `NatIPNames` and `NatIPs`?

```golang
NatIPs []NatIP `json:"natIPs,omitempty"`
```

```golang
type NatIP struct {
  IP *string `json:"ip,omitempty"`
  Name *string `json:"name,omitempty"`
}
```

Only one (ip or name) can be set at a time. WDYT?